kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ProxyConfigSecret.java

```diff
@@ -119,22 +120,23 @@ String generateProxyConfig(KafkaProxy primary,
     }
 
     @NonNull
-    private static LinkedHashMap<String, VirtualCluster> buildVirtualClusters(KafkaProxy primary, Context<KafkaProxy> context, List<Clusters> clusters) {
+    private static LinkedHashMap<String, VirtualCluster> buildVirtualClusters(KafkaProxy primary, Context<KafkaProxy> context, List<VirtualKafkaCluster> clusters) {
+        AtomicInteger clusterNum = new AtomicInteger(0);
```

Comment:
Why are we tracking the number of clusters? 

nit: `clusterNum` sounds like its for unique naming rather than being a counter of clusters.

Reply:
it's being driven by the current port allocation strategy, which should disappear at some point. The index of the virtual cluster determines its ports.